### PR TITLE
[_transactions2] Part 21: Release Notes and Exposing Transactions2

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Have understandable documentation on how to specify that we want to use transactions2

**Implementation Description (bullets)**:
- Add a docs section explaining how the config block works
- Call out transactions2 in release notes

**Testing (What was existing testing like?  What have you done to improve it?)**:
n/a, docs change

**Concerns (what feedback would you like?)**:
- Is advertising this this loudly a good idea?

**Where should we start reviewing?**: release-notes.rst

**Priority (whenever / two weeks / yesterday)**: ASAP

### CODE REVIEW

File: docs/source/configuration/internal-schemas.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+For a full list of the configurations available at this block, see
+`InternalSchemaConfig.java <https://github.com/palantir/atlasdb/blob/develop/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/InternalSchemaConfig.java>`__.
+
+Target Transactions Schema Version
+==================================
+
+.. warning::
+
+   _transactions2 is currently only supported for Cassandra and In-Memory KVSes.
+   In the case of Cassandra, _transactions2 is also only supported if you are using a version of Cassandra that supports
```
Comment (gmaretic) [inline docs/source/configuration/internal-schemas.rst]: If you don't have multi-pue, your Cassandra KVS is not going to be able to commit in `_transactions(1)` either, so seems like a superfluous comment

File: docs/source/configuration/internal-schemas.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- version 1, which variable-length encodes the start and commit timestamps and stores them in the ``_transactions``
+  table.
+- version 2, which variable-length encodes the start and commit timestamps following the
+  `TicketsEncodingStrategy <https://github.com/palantir/atlasdb/blob/develop/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/encoding/TicketsEncodingStrategy.java>`__,
+  storing them in the ``_transactions2`` table.
+
+If specified, this AtlasDB client will attempt to install the provided transaction schema version. This can be done in
+a live fashion without downtime, though it may take a while (in terms of AtlasDB timestamps) to take effect. If
+install is successful, a message will be logged indicating when the switch will take place. If needed, one can force
+the new version to be used by fast-forwarding the timestamp to that point.
```
Comment (gmaretic) [inline docs/source/configuration/internal-schemas.rst]: link to docs on how to do fast-forwarding?

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -50,16 +50,29 @@ develop
     *    - Type
          - Change
 
-    *    - |devbreak| |improved|
-         - The `TableMetadata` class has been refactored to use Immutables.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/3624>`__)
+    *    - |new|
```
Comment (gmaretic) [inline docs/source/release_notes/release-notes.rst]: Maybe note that this is now a beta feature?

File: docs/source/configuration/internal-schemas.rst (hunk 1 above)
Comment (jeremyk-91) [reply docs/source/configuration/internal-schemas.rst]: Good point, didn't think about this. Will remove this line.

File: docs/source/release_notes/release-notes.rst (hunk 3 above)
Comment (jeremyk-91) [reply docs/source/release_notes/release-notes.rst]: Added an explicit mention, do let me know if you think the wording should be stronger.

File: docs/source/configuration/internal-schemas.rst (hunk 2 above)
Comment (jeremyk-91) [reply docs/source/configuration/internal-schemas.rst]: Makes sense, added a link.

**Pull request merged**